### PR TITLE
DEP: remove `frechet_l` and `frechet_r`

### PR DESCRIPTION
**Pull request by rgommers**
These have been deprecated since 1.0.0, and this allows getting rid of doc build warnings and warning filters in many places.

### CODE REVIEW

File: scipy/stats/tests/common_tests.py [hunk 1]
```diff
@@ -312,9 +312,6 @@ def check_freezing(distfn, args):
 def check_rvs_broadcast(distfunc, distname, allargs, shape, shape_only, otype):
     np.random.seed(123)
     with suppress_warnings() as sup:
-        # frechet_l and frechet_r are deprecated, so all their
-        # methods generate DeprecationWarnings.
-        sup.filter(category=DeprecationWarning, message=".*frechet_")
```
Comment (peterbell10) [inline scipy/stats/tests/common_tests.py:317]: I think the entire context manager can go. Same for all the others I think.

Comment (WarrenWeckesser) [reply scipy/stats/tests/common_tests.py:317]: Yeah, if the only warning being suppressed is the frechet deprecation, the context manager can go.  If it helps, you can see where the context managers were added in https://github.com/scipy/scipy/pull/7838/files

Comment (rgommers) [reply scipy/stats/tests/common_tests.py:317]: Good catch, thanks.

**Review by ilayn — APPROVED** [review]

**Comment by ev-br** [comment]
Comments have been addressed, CI is green, merged. Thanks Ralf, all.

**Pull request merged**